lektor_index_pages/pruner.py

```diff
@@ -0,0 +1,38 @@
+'''
+Static collector for build-artifact urls.
+All non-tracked VPATH-urls will be pruned after build.
+'''
+from lektor.builder import Builder
+from lektor.reporter import reporter
+from lektor.utils import prune_file_and_folder
+
+_cache = set()
```

Comment:
The risk of concurrency issues is very low. I am not sure whether the `set.add()` method is atomic and I cant find the python source code for the definition of `set`. If it is not, we could add mutex around the add method. But I honstly think this is a non-issue. In the very rare occasion that two calls happen simultaneously and the python implementation of `add` "reads the value, adds the new value, and then writes the set", you would indeed miss a page. This page in turn would be pruned at the end of the build.

If old data remains in the set, thats fine. Worst case, the artifacts are not pruned. As is the current state. But the next time you start your lektor server they will be pruned normally because the set would be empty again.

Reply:
Oh, and if indeed a race condition should occur, it will automatically be fixed in the next build cycle.